pulp_container/app/authorization.py

```diff
@@ -92,32 +118,70 @@ def _split_into_encoded_groups(self, encoded_base32):
                 kid += char
         return kid
 
-    def determine_access(self, user, scope):
+    def determine_access(self):
         """
         Determine access permissions for a corresponding user.
 
         This method determines whether the user has a valid access permission or not.
-        The determination is based on role based access control. For now, the access
-        is given out to anybody because the role based access control is not implemented
-        yet.
-
-        Args:
-            user (str): A name of the user who is trying to access a registry.
-            scope (str): A requested scope.
+        The determination is based on role based access control.
 
         Returns:
             list: An intersected set of the requested and the allowed access.
 
         """
-        typ, name, actions = scope.split(":")
-        actions_list = actions.split(",")
-        permissions = {"pull"}
-        if user == "admin":
-            permissions.add("push")
-        permitted_actions = list(set(actions_list).intersection(permissions))
-        return {"type": typ, "name": name, "actions": permitted_actions}
-
-    def _generate_claim_set(self, issuer, issued_at, subject, audience, access):
+        typ, name, actions = self.scope.split(":")
+        actions = set(actions.split(","))
+
+        permitted_actions = set()
+        for action in actions:
+            has_permission = self.actions_permissions[action](name)
+            if has_permission:
+                permitted_actions.add(action)
```

Comment:
https://github.com/pulp/pulp_container/pull/203#discussion_r557507443

you can do a set comprehension here i think:
(Suggested change in the link)

Also why make it a set in the first place when all you do is to turn into a list?

Reply:
I thought it would be useful to use `set` here because we may receive a list of the same actions, i.e., `actions=pull,pull,push`. This way of permitting the actions should ignore duplicates.

Reply:
right. I did not see this, but you can never be sure about user input. ;)

Reply:
@lubosmj this should not be a concern since we are controlling this in the auth header https://github.com/pulp/pulp_container/blob/master/pulp_container/app/token_verification.py#L100
But i see an improvement here to return `pull,push` https://github.com/pulp/pulp_container/blob/master/pulp_container/app/token_verification.py#L34